docs/conf.in.py

```diff
@@ -270,8 +271,22 @@ def skip(app, what, name, obj, would_skip, options):
     return would_skip
 
 
+def escape_vertical_bars(app, what, name, obj, options, lines: List[str]):
+    # Note - Latex docs also contains Vertical Bar, so we
+    # apply this filtering only to specific classes.
+    if name in [
+            "open3d.data.RedwoodIndoorLivingRoom1",
+            "open3d.data.RedwoodIndoorLivingRoom2",
+            "open3d.data.RedwoodIndoorOffice1",
+            "open3d.data.RedwoodIndoorOffice2",
+    ]:
+        for index in range(len(lines)):
+            lines[index] = lines[index].replace("|", "\|")
+
+
 def setup(app):
     app.connect("autodoc-skip-member", skip)
+    app.connect("autodoc-process-docstring", escape_vertical_bars)
```

Comment:
The issue comes from vertical bars in python docs which are from tree command - http://open3d.org/docs/release/python_api/open3d.data.RedwoodIndoorLivingRoom1.html. This change fixes the warning but the output is in html paragraph tag and new lines get ignored. 

![Screenshot from 2023-08-12 18-26-51](https://github.com/isl-org/Open3D/assets/11634769/42a6d3bf-9b4c-449f-908f-b3e643cacfdc)

There was an alternate option to also remove output of tree command from docs, but I thought better keep tree output. Python docs are nicer, even if sphinx is not _yet_.

Reply:
@ssheorey Thanks for catching this. Learnt something new. Reverted my earlier postprocessing script and added suggested solution. Worked nicely for both sphinx and doxygen.